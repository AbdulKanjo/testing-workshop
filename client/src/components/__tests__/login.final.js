import React from 'react'
import {generate, render, Simulate} from 'client-test-utils'
import Login from '../login'

test('calls onSubmit with the username and password when submitted', () => {
  // Arrange
  const fakeUser = generate.loginForm()
  const handleSubmit = jest.fn()
  const {container, queryByLabelText, queryByText} = render(
    <Login onSubmit={handleSubmit} />,
  )

  const usernameNode = queryByLabelText('username')
  const passwordNode = queryByLabelText('password')
  const formNode = container.querySelector('form')
  const submitButtonNode = queryByText('submit')

  // Act
  usernameNode.value = fakeUser.username
  passwordNode.value = fakeUser.password
  Simulate.submit(formNode)

  // Assert
  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(handleSubmit).toHaveBeenCalledWith(fakeUser)
  expect(submitButtonNode.type).toBe('submit')
})

test('snapshot', () => {
  const {container} = render(<Login />)
  expect(container.firstChild).toMatchSnapshot()
})
