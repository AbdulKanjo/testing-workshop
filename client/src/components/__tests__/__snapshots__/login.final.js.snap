// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.css-0,
[data-css-0] {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0px;
  overflow: hidden;
  clip: rect(0,0,0,0);
  white-space: nowrap;
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  border: 0px;
}

.css-1,
[data-css-1] {
  background: white;
  height: 50px;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 30%;
  display: block;
  margin: 0 auto 10px auto;
  padding-left: 10px;
}

.css-1::-webkit-input-placeholder,
[data-css-1]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-1::-moz-placeholder,
[data-css-1]::-moz-placeholder {
  opacity: 0.5;
}

.css-1::-ms-input-placeholder,
[data-css-1]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-1::placeholder,
[data-css-1]::placeholder {
  opacity: 0.5;
}

.css-6,
[data-css-6] {
  font-size: 13px;
  font-family: Raleway,sans-serif;
  background: var(--green);
  padding: 10px 20px;
  display: block;
  margin: auto;
  color: white;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  cursor: pointer;
  transition: 0.5s;
  -webkit-transition: 0.5s;
  -moz-transition: 0.5s;
}

.css-6:hover,
[data-css-6]:hover {
  box-shadow: var(--shadowHover);
}

@media only screen and (max-width: 744px) {
  .css-1,
  [data-css-1] {
    width: 70%;
  }
}

<div>
  <form
    data-testid="login-form"
  >
    <label
      class="css-0"
      for="username-input"
    >
      Username
    </label>
    <input
      class="css-1"
      id="username-input"
      name="username"
      placeholder="Username..."
    />
    <label
      class="css-0"
      id="password-input"
    >
      Password
    </label>
    <input
      aria-labelledby="password-input"
      class="css-1"
      name="password"
      placeholder="Password..."
      type="password"
    />
    <button
      class="css-6"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
